app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabsIntegration.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray.syncedtabs
+
+import android.content.Context
+import androidx.lifecycle.LifecycleOwner
+import androidx.navigation.NavController
+import mozilla.components.browser.storage.sync.SyncedDeviceTabs
+import mozilla.components.feature.syncedtabs.SyncedTabsFeature
+import mozilla.components.feature.syncedtabs.storage.SyncedTabsStorage
+import mozilla.components.feature.syncedtabs.view.SyncedTabsView
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.observer.Observable
+import mozilla.components.support.base.observer.ObserverRegistry
+import org.mozilla.fenix.tabstray.TabsTrayAction
+import org.mozilla.fenix.tabstray.TabsTrayStore
+
+class SyncedTabsIntegration(
+    private val store: TabsTrayStore,
+    private val context: Context,
+    private val navController: NavController,
+    storage: SyncedTabsStorage,
+    accountManager: FxaAccountManager,
+    lifecycleOwner: LifecycleOwner,
+) : LifecycleAwareFeature,
+    SyncedTabsView,
+    Observable<SyncedTabsView.Listener> by ObserverRegistry() {
+
+    private val syncedTabsFeature by lazy {
+        SyncedTabsFeature(
+            context = context,
+            storage = storage,
+            accountManager = accountManager,
+            view = this,
+            lifecycleOwner = lifecycleOwner,
+            onTabClicked = {
+                // We can ignore this callback here because we're not connecting the adapter
+                // back to the feature. This works fine in other features, but passing the listener
+                // to other components in this case is annoying.
+            }
+        )
+    }
+
+    override var listener: SyncedTabsView.Listener? = null
```

Comment:
I don't see the `listener` assigned anywhere? Wondering if we need this and  `syncButtonBinding`?

Reply:
The listener is part of the component interface that I [mentioned here](https://github.com/mozilla-mobile/fenix/issues/21900). It made sense at one point to have a listener (interactor) tightly coupled with the view at the time, but it no longer makes sense so we should separate those two out.

Reply:
This listener was some weird black magic during the rewrite. The synced tabs data wasn't actually updating visually until I added the `Observable<SyncedTabsView.Listener> by ObserverRegistry()` line to the `SyncedTabsIntegration` class. I agree that this part of AC could probably be refactored as a larger part of the TabsTray rewrite.

Reply:
Probably good to file a followup isssue and drop some deets when it's still clear in your mind.

Reply:
https://github.com/mozilla-mobile/android-components/issues/11552

Feel free to add any edits to this